packages/eas-cli/src/utils/profiles.ts

```diff
@@ -8,19 +11,51 @@ type EasProfile<T extends ProfileType> = T extends 'build'
 export type ProfileData<T extends ProfileType> = {
   profile: EasProfile<T>;
   platform: Platform;
+  resourceClass: BuildResourceClass;
   profileName: string;
 };
 
+function getBuildResourceClassForPlatform({
+  resourceClass,
+  platform,
+}: {
+  resourceClass?: UserInputResourceClass;
+  platform: Platform;
+}): BuildResourceClass {
+  switch (platform) {
+    case Platform.ANDROID:
+      if (resourceClass === undefined || resourceClass === UserInputResourceClass.DEFAULT) {
+        return BuildResourceClass.AndroidDefault;
+      } else if (resourceClass === UserInputResourceClass.LARGE) {
+        return BuildResourceClass.AndroidLarge;
+      } else {
+        throw new errors.InvalidResourceClassError(`Invalid resource-class: ${resourceClass}`);
+      }
+    case Platform.IOS:
+      if (resourceClass === undefined || resourceClass === UserInputResourceClass.DEFAULT) {
+        return BuildResourceClass.IosDefault;
+      } else if (resourceClass === UserInputResourceClass.LARGE) {
+        return BuildResourceClass.IosLarge;
+      } else {
+        throw new errors.InvalidResourceClassError(`Invalid resource-class: ${resourceClass}`);
+      }
+    default:
+      throw new errors.InvalidResourceClassError(`Invalid resource-class: ${resourceClass}`);
+  }
```

Comment:
This is not really readable. It's better to have a map like:

```
const xyz: Record<Platform, Record<UserInputResourceClass, BuildResourceClass>> = {
  [Platform.ANDROID]: {
    [UserInputResourceClass.DEFAULT]: BuildResourceClass.AndroidDefault,
    // ..
  },
  // ..
};
```

And then try to access the appropriate value from map:
```
const aaa = xyz[platform]?.[userInputResourceClass];
if (!aaa) {
  throw new Error('blah blah');
}
```

My approach is better because:
- (!) If you ever add a new value to `Platform` or `UserInputResourceClass`, TS will complain that some mappings are missing in `xyz` map.
- The code is more straightforward.